transport/http/constants.go

```diff
@@ -56,4 +56,12 @@ const (
 	// service that can proxy for the destined service and perform
 	// application-specific routing.
 	RoutingDelegateHeader = "Rpc-Routing-Delegate"
+
+	// ApplicationStatusHeader is the HTTP response header used to express
+	// whether the response body represents an application error.
+	ApplicationStatusHeader = "Rpc-Status"
+
+	// ApplicationErrorStatus is the value for ApplicationStatusHeader for
+	// errant requests.
+	ApplicationErrorStatus = "error"
```

Comment:
It could be done—would need to set it initially and override in SetApplicationError.